tests/ut/backends/test_memory.py

```diff
@@ -225,3 +225,21 @@ def test_default_serializer(self):
 
     def test_parse_uri_path(self):
         assert SimpleMemoryCache().parse_uri_path("/1/2/3") == {}
+
+    @pytest.mark.asyncio
+    async def test_clear_cache_self_namespace(self):
+        namespace_a = "a"
+        namespace_b = "b"
+        cache_a = SimpleMemoryCache(namespace=namespace_a)
+        cache_b = SimpleMemoryCache(namespace=namespace_b)
+
+        await cache_a.set("foo", "bar")
+        val = await cache_a.get("foo")
+        assert "bar" == val
+
+        await cache_b.clear()
+        for key in cache_b._cache.keys():
+            assert not key.startswith(namespace_b)
```

Comment:
let's not use loops if the cache is empty this test passes. I would just go with an assert the _cache value and that's it

Reply:
Well, `cache_b._cache` can't be empty since it uses the same memory address for `_cache`, given how the backend is implemented `_cache` is a class variable.

I was trying to proof that all `b` keys were gone and the `a` ones were kept (before all of them would have been gone).

Do you have another suggestion on how to do this without a loop?

Reply:
For unit tests I would follow same approach is already used for testing the clear method
![image](https://user-images.githubusercontent.com/3578154/100773474-37295780-3401-11eb-8339-13f3697734a3.png)


then in acceptance I would do something like this:

```
await cache_a.set("a", 1)
await cache_b.set("b", 2)

await cache_b.clear()

assert cache_a.exists("a")
assert not cache_b.exists("b")
```